test/services/ScanUploadedFilesFlowSpec.scala

```diff
@@ -40,35 +41,74 @@ class ScanUploadedFilesFlowSpec extends UnitSpec with Matchers with GivenWhenThe
 
       val scanningService = mock[ScanningService]
-      Mockito
-        .when(scanningService.scan(any()))
-        .thenReturn(Future.successful(FileIsClean(S3ObjectLocation("bucket", "ID"))))
+      when(scanningService.scan(any())).thenReturn(Future.successful(FileIsClean(s3object)))
 
       val scanningResultHandler = mock[ScanningResultHandler]
-      Mockito.when(scanningResultHandler.handleScanningResult(any())).thenReturn(Future.successful(()))
+      when(scanningResultHandler.handleScanningResult(any())).thenReturn(Future.successful(SafeToContinue))
+
+      val instanceTerminator = mock[InstanceTerminator]
+
+      val queueOrchestrator =
+        new ScanUploadedFilesFlow(queueConsumer, parser, scanningService, scanningResultHandler, instanceTerminator)
+
+      When("the orchestrator is called")
+      Await.result(queueOrchestrator.run(), 30 seconds)
+
+      Then("the queue consumer should poll for messages")
+      verify(queueConsumer).poll()
+
+      And("scanning result handler is called")
+      verify(scanningResultHandler).handleScanningResult(FileIsClean(s3object))
+
+      And("successfully processed messages are confirmed")
+      verify(queueConsumer).confirm(validMessage)
+    }
+
+    "terminate instance if virus found" in {
+      Given("there are only valid messages in a message queue")
+      val validMessage = Message("ID", "VALID-BODY", "RECEIPT-1")
+      val s3object     = S3ObjectLocation("bucket", "ID")
+
+      val queueConsumer = mock[QueueConsumer]
+      when(queueConsumer.poll()).thenReturn(List(validMessage))
+      when(queueConsumer.confirm(any())).thenReturn(Future.successful(()))
+
+      val scanningService = mock[ScanningService]
+      when(scanningService.scan(any())).thenReturn(Future.successful(FileIsInfected(s3object, "Virus name")))
+
+      val scanningResultHandler = mock[ScanningResultHandler]
+      when(scanningResultHandler.handleScanningResult(any())).thenReturn(Future.successful(ShouldTerminate))
+
+      val instanceTerminator = mock[InstanceTerminator]
+      when(instanceTerminator.terminate()).thenReturn(Future(()))
 
       val queueOrchestrator =
-        new ScanUploadedFilesFlow(queueConsumer, parser, scanningService, scanningResultHandler)
+        new ScanUploadedFilesFlow(queueConsumer, parser, scanningService, scanningResultHandler, instanceTerminator)
 
       When("the orchestrator is called")
       Await.result(queueOrchestrator.run(), 30 seconds)
 
       Then("the queue consumer should poll for messages")
-      Mockito.verify(queueConsumer).poll()
+      verify(queueConsumer).poll()
 
       And("scanning result handler is called")
-      Mockito
-        .verify(scanningResultHandler)
-        .handleScanningResult(FileIsClean(S3ObjectLocation("bucket", "ID")))
+      verify(scanningResultHandler)
+        .handleScanningResult(FileIsInfected(s3object, "Virus name"))
 
       And("successfully processed messages are confirmed")
```

Comment:
label says And("successfully processed messages are confirmed") but there is no code that confirms it